core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/SynapseExtension/SynapseExtensionUtilities.scala

```diff
@@ -0,0 +1,312 @@
+package com.microsoft.azure.synapse.ml.nbtest.SynapseExtension
+
+import com.microsoft.azure.synapse.ml.Secrets
+import com.microsoft.azure.synapse.ml.build.BuildInfo
+import com.microsoft.azure.synapse.ml.core.env.FileUtilities
+import com.microsoft.azure.synapse.ml.io.http.RESTHelpers
+import com.microsoft.azure.synapse.ml.io.http.RESTHelpers._
+import com.microsoft.azure.synapse.ml.nbtest.SynapseExtension.Models._
+import com.microsoft.azure.synapse.ml.nbtest.SynapseUtilities
+import org.apache.http.client.config.RequestConfig
+import org.apache.http.client.entity.UrlEncodedFormEntity
+import org.apache.http.client.methods._
+import org.apache.http.entity.StringEntity
+import org.apache.http.message.BasicNameValuePair
+import spray.json._
+
+import java.io.File
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future, TimeoutException, blocking}
+import scala.sys.process._
+
+object SynapseExtensionUtilities {
+  import SynapseJsonProtocol._
+
+  val TimeoutInMillis: Int = 30 * 60 * 1000
+
+  val SSPHost: String = "https://wabi-daily-us-east2-redirect.analysis.windows.net"
+  val WorkspaceId: String = "8f02ac2a-92cb-4f52-975e-4d0fa4a5cafa"
+
+  val BaseUri: String = s"$SSPHost/metadata"
+  val ArtifactsUri: String = s"$BaseUri/workspaces/$WorkspaceId/artifacts"
+
+  val TenantId: String = "pbidaily.onmicrosoft.com"
+  val AadAccessTokenResource: String = "https://analysis.windows.net/powerbi/api"
+  val AadAccessTokenClientId: String = "1950a258-227b-4e31-a9cf-717495945fc2"
+
+  val Folder: String = s"build_${BuildInfo.version}/synapseextension/notebooks"
+  val StorageAccount: String = "mmlsparkbuildsynapse"
+  val StorageContainer: String = "synapse-extension"
+
+  lazy val AccessToken: String = getAccessToken
+
+  def getStorageOAuthToken: String = {
+    val tokenFile = FileUtilities
+      .join(BuildInfo.baseDirectory.getParent,
+        "core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/SynapseExtension/storagetoken")
+      .getCanonicalFile
+    FileUtilities.readFile(tokenFile)
+  }
+
+  def createSJDArtifact(path: String): String = {
+    createSJDArtifact(path, "SparkJobDefinition")
+  }
+
+  def updateSJDArtifact(path: String, artifactId: String, lakehouseId: String): Artifact = {
+    val eTag = getETagFromArtifact(artifactId)
+
+    val excludes: String = "org.scala-lang:scala-reflect," +
+      "org.apache.spark:spark-tags_2.12," +
+      "org.scalactic:scalactic_2.12," +
+      "org.scalatest:scalatest_2.12," +
+      "org.slf4j:slf4j-api"
+      //"io.netty:netty-tcnative-boringssl-static-2.0.43.Final-," +
+    val reqBody: String =
+      s"""
+         |{
+         |  "workloadPayload": "{
+         |    \\"ExecutableFile\\": \\"$path\\",
```

Comment:
This is an odd payload with the double quoted strings, you sure this is the format? If so perhaps break this into two cars to avoid the double quoted strings

Reply:
Are you talking about the escape sequence here? Yeah it's an odd format indeed, but the "Workloadpayload" object itself is it's own serialized json string and it only is accepted with the double quotes escaped...
![image](https://user-images.githubusercontent.com/8703669/201179147-6f4e6dcb-43e6-4f74-af95-777deac73e7c.png)

I can certainly separate this into its own weird thing

Reply:
Nvm I figured it out. Nobody will have to look at this ugliness

Reply:
We already have! Gaah! My eyes! :)